src/components/select/index.tsx

```diff
@@ -0,0 +1,135 @@
+            selectedCity: 'Vilnius',
+            selectedDate: new Date(),
+            showMovies: false,
+            movies: [],
+        };
+
+        this.searchForMovies = this.searchForMovies.bind(this);
+    }
+
+    componentDidMount() {
+        this.formatDate(this.state.selectedDate)
+    }
+
+    showCitiesToSelect() {
+        let cityOptions = [];
+        for (let i = 0; i < this.state.cities.length; i++) {
+            cityOptions.push(<option key={i}>{this.state.cities[i]}</option>)
+        }
+        return cityOptions;
+    }
+
+    handleDateChange = (date: Date): void => {
+        this.setState({
+            selectedDate: date,
+            showMovies: false
+        });
+        this.formatDate(date);
+    };
+
+    selectCity(event: string): void {
+        this.setState({
+            selectedCity: event,
+            showMovies: false,
+        });
+    }
+
+    formatDate(date: Date): string {
+        let options = { month: 'long', day: 'numeric' }
+        this.formattedDate = date.toLocaleDateString('en-US', options);
+        return this.formattedDate;
+    }
+
+    searchForMovies() {
+        this.setState({ showMovies: true });
+        this.getMoviesByDay(this.state.selectedDate);
+    }
+
+    getMoviesByDay(day: Date) {
+        getMovies(day).then(data => {
+            data.forEach(movieDate => {
+                movieDate.date = new Date(movieDate.date);
+                if (isSameDay(movieDate.date, day)) {
+                    this.setState({ movies: movieDate.movies })
+                }
+            })
+            console.log(this.state.movies)
+            return this.state.movies;
+        });
+    }
+
+    // async displayMovies() {
+    //     let movieLi = []
+    //     if (this.state.movies !== null) {
+    //         for (let movie of this.state.movies) {
+    //             movieLi.push(movie.title)
+    //         }
+    //         this.setState({movieList: movieLi})
+    //     } else {
+    //         console.log('Ups')
+    //     }
+    // }
+
+    render() {
+        return (
+            <div>
+                <div className="select">
+                    <select className="cityToSelect" onChange={event => this.selectCity(event.target.value)}>
+                        {this.showCitiesToSelect()}
+                    </select>
+
+                    <DatePicker className="dateToSelect"
+                        selected={this.state.selectedDate}
+                        onChange={this.handleDateChange}
+                        minDate={new Date()}
+                        dateFormat="yyyy-MM-dd"
+                    />
+
+                    <button className="search-btn" onClick={this.searchForMovies}>Search</button>
+                </div>
+                {this.state.showMovies ?
+                    <div>
+                        <div className="movie-list-title">
+                            <h1>On {this.formattedDate} in {this.state.selectedCity} we show</h1>
```

Comment:
Fields, rendered in JSX, should be in state of the component. In this case, it could be rewritten to
```suggestion
                            <h1>On {this.formatDate(this.state.selectedDate)} in {this.state.selectedCity} we show</h1>
```